packages/angular/cli/models/architect-command.ts

```diff
@@ -168,14 +170,41 @@ export abstract class ArchitectCommand<
     return await this.runArchitectTarget(options);
   }
 
-  protected async runSingleTarget(targetSpec: TargetSpecifier, options: string[]) {
+  protected async runBepTarget<T>(
+    command: string,
+    configuration: BuilderConfiguration<T>,
+    buildEventLog: string,
+  ): Promise<number> {
+    // Send start
+    appendEvent(
+      buildEventLog,
+      BepGenerator.createBuildStarted(command),
+    );
+
+    let last = 1;
+    await this._architect.run(configuration, { logger: this.logger }).forEach(event => {
+      last = event.success ? 0 : 1;
+
+      appendEvent(
+        buildEventLog,
+        BepGenerator.createBuildFinished(last),
```

Comment:
I'm a bit confused, if this happens in a loop, does it mean we produce more than one buildfinished event? I'd expect there is always a single one

Reply:
In watch/dev-server mode there are potentially multiple builds per command invocation.